### ISSUE TITLE
'promises' is not exported by __vite-browser-external

### ISSUE THREAD

**Issue by makeryangcom** [ollama/ollama-js#76]
```
../../release/dist/preload/browser.cjs   0.14 KiB / gzip: 0.12 KiB
vite v3.2.8 building for production...
✓ 2006 modules transformed.
'promises' is not exported by __vite-browser-external, imported by node_modules/ollama/dist/index.mjs
file: D:/ChatDesktop/node_modules/ollama/dist/index.mjs:2:13
1: import { O as Ollama$1, h as head, p as post } from './shared/ollama.be8fd0da.mjs';
2: import fs, { promises, createReadStream } from 'fs';
                ^
3: import { resolve, join, dirname } from 'path';
4: import { createHash } from 'crypto';
```

**Comment by m0o0scar** [comment]
@makeryangcom could you please explain your issue with some more detail? how you encoutered this error? what're the steps to reproduce?

**Comment by makeryangcom** [comment]
I have solved this problem; it can be avoided by not using the browser.

**Comment by tang2087** [comment]
You can use polyfills for browser:
https://kontext.tech/article/1375/resolve-promises-is-not-exported-by-vite-browser-external

**Comment by cvl01** [comment]
For anyone coming across this, the library has been updated with a fix for this in #47. 

Change your import to this, to only import browser-compatible functionality. 

```js
import ollama from 'ollama/browser'
```
